drizzlepac/haputils/mvm_quality_graphics.py

```diff
@@ -546,6 +554,243 @@ def wcs_graphics_driver(storage_filename, output_base_filename='', display_plot=
+            for expo in range(len(all_expo_X)):
+                wcs_value = all_wcs[expo]
+                x_corners = all_expo_X[expo]
+                y_corners = all_expo_Y[expo]
+                shape = all_shape[expo]
+                if layer_mask is None:
+                    layer_mask = np.zeros(shape, dtype=np.int32)
+                    cnt_layer_mask = np.zeros(shape, dtype=np.uint8)
+
+                expo_mask = None
+                # Loop over each chip in an exposure
+                for ichip in range(len(all_expo_X[0])):
+                    polygon = list(zip(x_corners[ichip], y_corners[ichip]))
+                    naxis1, naxis2 = shape
+
+                    # Take care to reverse the dimensions to match those of Numpy
+                    #
+                    # Be careful with the mode - the layer_mask can have a max value
+                    # of less than 2^21 (2^20 + other), and the cnt_layer_mask is the
+                    # number of exposures in that portion of the mask which we will
+                    # assume is no larger than 2^16.
+                    cnt_img = Image.new("L", (naxis2, naxis1), color=0)
+                    ImageDraw.Draw(cnt_img).polygon(polygon, outline=1, fill=1)
+
+                    # Update the image counter mask
+                    cnt_expo_mask = np.array(cnt_img).astype(np.uint8)
+                    cnt_layer_mask += cnt_expo_mask
+                    cnt_all_layers_mask += cnt_expo_mask
+
+                    # Set filled portion of the exposure mask to the WCS 2^n value
+                    # and update the bitwise_OR mask
+                    layer_img = Image.new("I", (naxis2, naxis1), color=0)
+                    ImageDraw.Draw(layer_img).polygon(polygon, outline=1, fill=1)
+                    expo_mask = np.array(layer_img).astype(np.int32)
+                    expo_mask *= wcs_value
+                    np.bitwise_or(layer_mask, expo_mask, out=layer_mask, dtype=np.int32)
+                    np.bitwise_or(all_layers_mask, expo_mask, out=all_layers_mask, dtype=np.int32)
+
+            # Compute statistics for a specific layer
+            stats = compute_overlap_stats(layer_mask, layer)
+            layer_stats += stats
+
+        # Compute statistics for the composite of all the layers
+        stats = compute_overlap_stats(all_layers_mask, "All Layers")
+        layer_stats += stats
+        max_color = cnt_all_layers_mask.max() + 1
+
+        # Resample the output footprint overlay image of all the layers in the skycell
+        # as the original image is large and deep (Based on code in astrometric_utils.py.)
+        scale = 8
+        yend = all_layers_mask.shape[0] % scale
+        xend = all_layers_mask.shape[1] % scale
+        yend = -1 * yend if yend > 0 else None
+        xend = -1 * xend if xend > 0 else None
+        new_shape = (all_layers_mask.shape[0] // scale, all_layers_mask.shape[1] // scale)
```

Comment:
Would it be useful to also shrink the 'cnt_all_layers_mask' as well?  It may make the final HTML file render faster.

Reply:
Indeed.  However, we only want to slice the image to make it smaller.  We do not want to touch the image pixel "intensities".  This has been done in update 555997b.